contracts/mapping-token/contracts/v2/Backing.sol

```diff
@@ -0,0 +1,268 @@
+        outboundLanes[bridgedLanePosition] = outboundLane;
+        emit NewOutBoundLaneAdded(bridgedLanePosition, outboundLane);
+    }
+
+    /**
+     * @notice reigister new erc20 token to the bridge. Only owner can do this.
+     * @param bridgedLanePosition the bridged lane positon, this register message will be delived to this lane position
+     * @param token the original token address
+     * @param name the name of the original token
+     * @param symbol the symbol of the original token
+     * @param decimals the decimals of the original token
+     */
+    function registerErc20Token(
+        uint32 bridgedLanePosition,
+        address token,
+        string memory name,
+        string memory symbol,
+        uint8 decimals
+    ) external payable onlyOwner {
+        require(registeredTokens[token] == false, "Backing: token has been registered");
+
+        address outboundLane = outboundLanes[bridgedLanePosition];
+        require(outboundLane != address(0), "cannot find outboundLane to send message");
+        bytes memory newErc20Contract = abi.encodeWithSelector(
+            IMappingTokenFactory.newErc20Contract.selector,
+            address(this),
+            ERC20_TOKEN_TYPE,
+            token,
+            thisChainName,
+            name,
+            symbol,
+            decimals
+        );
+        uint256 fee = IFeeMarket(feeMarket).market_fee();
+        require(msg.value >= fee, "not enough fee to pay");
+        uint64 nonce = IOutboundLane(outboundLane).send_message{value: fee}(remoteMappingTokenFactory, newErc20Contract);
+        uint256 messageId = encodeMessageId(bridgedLanePosition, nonce);
+        registerMessages[messageId] = token;
+        if (msg.value > fee) {
+            require(payable(msg.sender).send(msg.value - fee), "transfer back fee failed");
+        }
+        emit NewErc20TokenRegistered(messageId, bridgedLanePosition, token);
+    }
+
+    /**
+     * @notice lock original token and issuing mapping token from bridged chain
+     * @dev maybe some tokens will take some fee when transfer
+     * @param bridgedLanePosition the bridged lane positon, this issuing message will be delived to this lane position
+     * @param token the original token address
+     * @param recipient the recipient who will receive the issued mapping token
+     * @param amount amount of the locked token
+     */
+    function lockAndRemoteIssuing(
+        uint32 bridgedLanePosition,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external payable whenNotPaused {
+        require(registeredTokens[token], "Backing: the token is not registed");
+
+        uint256 balanceBefore = IERC20(token).balanceOf(address(this));
+        require(IERC20(token).transferFrom(msg.sender, address(this), amount), "transfer tokens failed");
+        uint256 balanceAfter = IERC20(token).balanceOf(address(this));
+        uint256 newAmount = balanceAfter - balanceBefore;
```

Comment:
Can we handle these details in a different token category, and then simply do the transfer here?

Reply:
https://github.com/OffchainLabs/arbitrum/blob/3d4365ab26f86b62d12c075dc86062799033698d/packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/gateway/L2CustomGateway.sol#L54

Reply:
Looks dangerous if not use checked_sub, how about only support tokens which are normal.
And for safety concern, it will require Wormhole DAO to add the token into white list even if it is registered/proposal.

I would assume Wormhole DAO will not remove white list, but it will ultimately relay on the decision of Wormhole DAO.

```
require(balanceBefore.checked_add(amount) == balanceAfter, "....")
```

Reply:
the register is onlyOwner, we can't process all unnormal tokens. The rebase token is not allowed to be registered.

Reply:
Balance checking is necessary, but is there a better way to solve this problem, because there is no different processing logic for different tokens.

Reply:
`require(balanceBefore.checked_add(amount) == balanceAfter, "....")` can solve this problem. But if so, a large number of tokens with fee when transferred will not allowed to be registered.

Reply:
I think for now, `require(balanceBefore.checked_add(amount) == balanceAfter, "....")` the check is probably better, because the mapping token does not have an implementation of the token with fee when transferred.